MainWindow.xaml.cs

```diff
@@ -446,64 +485,77 @@ private void CancelDumping()
         /// </summary>
         private void EnsureDiscInformation()
         {
+            var systemTuple = cmb_SystemType.SelectedItem as Tuple<string, KnownSystem?>;
+            var discTypeTuple = cmb_DiscType.SelectedItem as Tuple<string, DiscType?>;
+
             // If we're on a separator, go to the next item
-            var tuple = cmb_DiscType.SelectedItem as Tuple<string, KnownSystem?, DiscType?>;
-            if (tuple.Item2 == null && tuple.Item3 == null)
+            if (systemTuple.Item2 == null)
+                systemTuple = cmb_SystemType.Items[++cmb_SystemType.SelectedIndex] as Tuple<string, KnownSystem?>;
+
+            //TODO: a specific KnownSystem to manage Custom would be better
+            bool isCustom = systemTuple.Item1 == "Custom Input";
+            var selectedSystem = systemTuple.Item2;
+            var selectedDiscType = discTypeTuple != null ? discTypeTuple.Item2 : DiscType.NONE;
+
+            // No system chosen, update status
+            if (selectedSystem == KnownSystem.NONE)
             {
-                cmb_DiscType.SelectedIndex++;
```

Comment:
Why did you separate out the KnownSystem.NONE from the rest of the switch statement? I don't see any advantage to doing so

Reply:
The switch is over DiscType, not on KnownSystem, we're comparing apples with oranges here.